src/Analyzers/Core/CodeFixes/Formatting/FormattingCodeFixProvider.cs

```diff
@@ -30,8 +30,8 @@ public sealed override ImmutableArray<string> FixableDiagnosticIds
         /// them acting on an error reported in code, and can be computed fast as it only uses syntax not semantics.
         /// It's also the 8th most common fix that people use, and is picked almost all the times it is shown.
         /// </summary>
-        private protected override CodeActionRequestPriority ComputeRequestPriority()
-            => CodeActionRequestPriority.High;
+        private protected override CodeActionRequestPriorityInternal ComputeRequestPriorityInternal()
```

Comment:
So, this API/override is still not available in the code style layer. This means the formatting analyzer/fixer when executed from the CodeStyle NuGet package would still be clamped down to normal priority. I am wondering if there is some way we can incorporate that support in this PR.

Reply:
I am wondering if the following implementation could work here:

1. Do not add the parallel XXXInternal APIs that are being added in this PR, instead just have the public ones, also exposing the `High` field from these enums
2. Have our IDE code fixers/refactorings add a special custom tag to indicate these are first party ones: https://github.com/dotnet/roslyn/blob/ef55b116221d058f68eee08923315959aef3b314/src/Workspaces/Core/Portable/CodeActions/CodeAction.cs#L107-L111
3. Clamp the `High` priority values down to `Normal` priority for providers and actions that don't have these special custom tag.

Agreed that this opens a small backdoor for third party providers to now add this special custom tag themselves and trick us into thinking it's a first party provider, but I think such providers will be rare, if any. On the positive side though, we avoid all this overengineering with parallel Internal and Public APIs, and can finally have our first party providers shipping in CodeStyle NuGet package to also be able to run at high priority when appropriate.

Reply:
Sounds good. I'll do that!

Reply:
Slightly confused though. If this is not available, why would this Compile

Reply:
Ignore that. I see the ifdef :-)